### PR TITLE
Fix const error when compiling in clang.

### PR DESCRIPTION
**Pull request by cianciosa**
## Proposed changes

This fixes the second bug in https://github.com/project-asgard/asgard/issues/387. The bug arrises because `plan_` contains elements that are marked `const` and the line `plan_ = new_plan` tries to overwrite them which would violate the `const` property. This pr works around that by deleting the old elements of `plan_` first then inserting the elements of `new_plan`

## What type(s) of changes does this code introduce?
_Put an `x` in the boxes that apply._

- [x] Bugfix
- [ ] New feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

## What systems has this change been tested on?
Mac OS and Linux. Running the unit tests locally result in several test failures of unknown origin.

## Checklist

- [x] this PR is up to date with current the current state of 'develop'
- [ ] code added or changed in the PR has been clang-formatted
- [ ] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [ ] documentation has been added (if appropriate)

### CODE REVIEW

File: src/adapt.cpp [hunk 1]
```diff
@@ -312,7 +312,8 @@ fk::vector<P> distributed_grid<P>::refine_elements(
   auto const new_plan = get_plan(get_num_ranks(), table_);
   auto const remapper = remap_for_addtl(table_.size() - added);
   auto const y        = redistribute_vector(x, plan_, new_plan, remapper);
-  plan_               = new_plan;
```
Comment (quantumsteve) [inline src/adapt.cpp:315]: Please add a comment about why we're doing this.

File: src/adapt.cpp [hunk 2]
```diff
@@ -338,7 +339,9 @@ fk::vector<P> distributed_grid<P>::remove_elements(
   auto const new_plan = get_plan(get_num_ranks(), table_);
   auto const remapper = remap_for_delete(all_remove_indices, table_.size());
   auto const y        = redistribute_vector(x, plan_, new_plan, remapper);
-  plan_               = new_plan;
+
```
Comment (quantumsteve) [inline src/adapt.cpp:346]: Please add a comment about why we're doing this.

**Review by quantumsteve — CHANGES_REQUESTED** [review]
Please add a code comment and squash your commits.

**Comment by cianciosa** [comment]
🤬🤬🤬 Every time I try to squash the commits, it just keeps adding more.

**Review by quantumsteve — APPROVED** [review]

**Pull request merged**